tests/php/test-class-customize-snapshot-manager.php

```diff
@@ -729,4 +729,45 @@ public function test_replace_customize_link() {
 		$parsed_preview_url = wp_parse_url( $query_params['url'] );
 		$this->assertArrayNotHasKey( 'query', $parsed_preview_url );
 	}
+
+	/**
+	 * Test filter_user_has_cap.
+	 *
+	 * @covers \CustomizeSnapshots\Customize_Snapshot_Manager::filter_user_has_cap()
+	 */
+	function test_filter_user_has_cap() {
+		$this->mark_incompatible();
+		$post_id = $this->factory()->post->create( array(
+			'post_type' => 'post',
+			'post_status' => 'auto-draft',
+		) );
+
+		$uuid = wp_generate_uuid4();
+		$this->manager->post_type->save( array(
+			'uuid' => $uuid,
+			'data' => array(
+				'post[post][' . $post_id . ']' => array(
+					'value' => array(
+						'post_status' => 'publish',
+					),
+				),
+			),
+			'status' => 'draft',
+		) );
+
+		$this->manager->current_snapshot_uuid = $uuid;
+		$caps = array(
+			'0' => 'read_others_posts',
+		);
+		$args = array(
+			'0' => 'read_post',
+			'2' => $post_id,
+		);
+		$allcaps = $this->manager->filter_user_has_cap( array(), $caps, $args );
```

Comment:
Instead of using the `filter_user_has_cap` method directly, should this instead use `current_user_can` to ensure that the filter is applying?